### DISCUSSION TITLE
Error on upgrading to version 10 and translations problems

### DISCUSSION THREAD

**Discussion by CultVonnegut** [cerb/cerb-release#21] [Q&A] [answered]
Hello,
First of all I don't know if this is the right place to ask this, so apologies in advance.
We are trying to update an old instance of cerb (v5) to the last version 10.
I downloaded the git repo, followed the instructions on: https://cerb.ai/docs/upgrading/ and installed all dependencies for php7.4 if we go to the https://sitename/install all requisites checks are validated and after accepting the license it detects it is an upgrade so it offers a link to the main address https://sitename/ but it redirects to https://sitename/update/locked but we get an error from apache:
Not Found
The requested URL was not found on this server.
So we are stuck on this loop.
Thank you very much in advance.
Kind regards

Solution: https://github.com/cerb/cerb-release/discussions/21#discussioncomment-2789412

**Comment by jstanden** [comment]
Sure! This is the right place if you don't share any private details. Otherwise you can email [team@cerb.ai](mailto:team@cerb.ai) whether your license is current or not.

This sounds like an Apache config issue handing off to mod_php or php-fpm.

Can you verify that even a simple `phpinfo.php` page works in your web path? Like:

```php
<?php
phpinfo();
```

**Reply by CultVonnegut** [reply]
Hi, thanks for you answer.
We created an info.php with the phpinfo function and it is working alright. So I don't know if it is a permission issue, but I have changed the owner to the apache user and set the right permissions but, I am missing something.
Is there any other place you could point to?
Thanks a lot

**Comment by CultVonnegut** [comment, marked as answer]
Hi again
A little bit more of information. I have repeated the process but I haven't followed the upgrade process, instead of that I have downloaded the repo again and put the previous database so it is upgraded but when entering in the url it appears:
[ERROR] Can't initialize the Devblocks cache.

Kind regards

**Reply by jstanden** [reply]
If your filesystem paths changed, remove the ./storage/tmp/cache--* files.

If you were using Redis or Memcached for the cache before, make sure those PHP extensions are installed.

**Reply by CultVonnegut** [reply]
Thanks again, we have the previous data on a new server in a new directory. I have deleted the storage/tmp/devblocks-cache--* but problem persists.
If I put in the framework.config.php a clean database of the version 10 it shows the login page corectly after the install process, if I put the old database after the install process and upgrade detection it is redirected to a not found index.php.
We need to migrate and old server to a new one and we would like to migrate all data (16gb)
Thanks again jstanden!!!
Kind regards!

**Reply by jstanden** [reply]
It sounds like it's possibly trying to redirect to `/install/` when it detects the database is empty. The 404 is from that path not existing as it's normally removed post-install.

Have you tested the MySQL host/user/password/database from your old `framework.config.php` on the new server? I recommend testing a connection with the `mysql` CLI first.

**Reply by CultVonnegut** [reply]
Hi jstanden sorry for the troubles and thanks for support.
I have just realized that the colleague that was managing this had installed mysql v.8, I guess it could affect, I don't know. I am reverting the server to an initial state and installing mysql v.5.7.
Thanks a lot
Have a nice day.

**Reply by jstanden** [reply]
Cerb 10.0 supports MySQL 8.0, but the recent versions of MySQL also change user authentication by default:
https://medium.com/@crmcmullen/how-to-run-mysql-8-0-with-native-password-authentication-502de5bac661

You just need to make sure you're using native passwords for the Cerb user.

Downgrading to MySQL 5.7 for now is fine too. My dev environment uses 8.0.

**Reply by CultVonnegut** [reply]
Hi again! I created a new DB user and password.
I have reconfigured the server with mysql 5.7, php 7.4, downloaded the repo with git and dumped the old database to the new server. I follow the install process, it detects the old database and asks for upgrading, but now it is stuck on a page with the message:
"Another administrator is currently running an update. Please wait..." It is there since yesterday and I don't know what to do.
In the error.log from apache I got this:

"[Wed May 05 08:08:25.814380 2021] [proxy_fcgi:error] [pid 5759] [client 172.x.x.x:41312] AH01071: Got error 'PHP message: [1054] Unknown column 'refreshed_at' in 'field list' ::SQL:: UPDATE devblocks_session SET updated=1620194905, refreshed_at=1620194905, session_data='csrf_token|s:128:\\"KF98YV97MFYUWCV1V5F4WTYXK1F8ZW33Y32BH55DT2Q9VKBJQ9UPSHPB92DZKNHJSFYMFV763KKZUMN2KL7CGA2JVR19KLBG232VCP65MF71EKURD235AHZW76FDJ96T\\";', user_id=0, user_ip='172.x.x.x', user_agent='Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:82.0) Gecko/20100101 Firefox/82.0' WHERE session_token='51t5vb3a85n5gknsetqivmhda6'', referer: https://xxxxxxx.xxxx.site.com/index.php/update/locked"

Thank you.

**Reply by Hildy** [reply]
Delete the lock file (on disk) and then reload the update URL and let us know what you see then.

**Reply by jstanden** [reply]
Agreed!

Remove `./storage/tmp/cerb_update_lock` (if still locked), then remove `./storage/version.php`. Cerb will prompt you to patch again.

**Reply by CultVonnegut** [reply]
Thank you very much for your responses, after tinkering a bit with it I went back to the root folder in the webserbver and the login page appeared!! so exciting. I got in with my old user and the imported data is there.
Though there are some odd things like when I go to the top right menu where the user name appears when I click on it there is a rolling menu which shows:

header.myprofile
header.my_card
settings
notifications
common.activiry_log
salir
header.signoff.all.my
Any clue, do I need to change some strings or something in any file?

Thanks Hildy and jstanden.
Kind regards

**Reply by CultVonnegut** [reply]
One more thing, I didn't copy the old storage folder, what would be the process to copy so it doesn't mess with the new installation.
Thanks.

**Reply by jstanden** [reply]
On the worker menu, it sounds like the translations aren't in the database. Those are read from the disk during an update. You can rebuild the translations by repeating the steps above (remove `./storage/version.php` then let Cerb update again).

On copying storage, that could be your issue above too, if the permissions aren't set properly.

If you're worried about conflicts by replacing the whole `./storage` directory (which is fine to do), you can copy these specific paths between installs:

* `./storage/_master.key` (if exists; used to encrypt connected services/accounts)
* `./storage/attachments/`
* `./storage/classes/` (if exists; used by custom records)
* `./storage/context_avatar/`
* `./storage/logo` (if exists)

**Reply by CultVonnegut** [reply]
Good Morning!
I have copied the storage folder and deleted the version.php and then do the upgrade process again, but the weird literal still appears, even on login page the login button shows Common.Continue.
Thank you very much for all!

**Reply by CultVonnegut** [reply]
Hi @jstanden,
We found where was the problem, we had a pretty old installation, but it seems the update process have a problem with the database so old.
Translation id had to convert to autoincrement, but it had an issue because for some reason there was an index id with value 0, and even altering table by hand to make it autoincrement launched an error. Something about the index. So we deleted the index id 0 and launched the update and went well, no more errors with translations after the update.
Can close the ticket and mark as solved

*Edit for clarification

**Reply by jstanden** [reply]
Hi @bodypheo! Thanks for letting me know. That will be useful in case the same issue comes up again.

5.x to 10.x is a pretty big jump, so let me know if there's anything I can help with.
